### DISCUSSION TITLE
Quais notebooks tem bom custo-benefício para web devs?

### DISCUSSION THREAD

**Discussion by matheus-rosin** [frontendbr/forum#1285] [Perguntas e Respostas]
Olá!

Ainda não tive contato com muitos notebooks, então no caso de comprar um fico com dúvidas.

Gostaria que compartilhassem de suas experiências e do que sabem a respeito, desde RAM até touchpad.

**Comment by klarkc** [comment]
Lenovo sempre tem um bom custo x hardware, mas acho os da Dell mais leves e resistentes (gosto da linha de ultrabooks vostro). Fora isso pegue 16GB de ram, é sério, é irritante trabalhar com 8GB só. E se puder um SSD.

**Comment by fxcosta** [comment]
Meu principal conselho, indo além de marcas e modelos, é pegar algo sempre além do que você imagina ideal. Sei que pode ser complicado e talvez você realmente não precise de algo a mais do que você imagina, mas no meu caso, quando comprei meu notebook (um i3, não me pergunte porque peguei um desses mesmo tendo sido a 3 anos atrás), só trabalhava com web.

E de boa uma config mais baixa se tu vai trabalhar com um xampp, um react ali, enfim, projetos web convencionais. O problema está sendo agora: trabalhar com react native é pesado se vc pensa em usar emulador, trabalhar com unity3d com modelos de até 1gb, o próprio nodejs pra rodar algumas coisas da uma capengada... Enfim, não vá nessa ideia de pegar somente o essencial. Pegue ao menos uma coisa além. 

Quanto a configurações e modelos, não posso opinar mto, só tive um notebook em toda minha vida. Um Acer. Acho um bom custo benefício. Não manjo quanto ao suporte. Mas hoje eu já pegaria no mínimo um I5, no mínimo 8gb de ram, placa de vídeo dedicada, 1 tb de HD e o touchpad q aceite múltiplos toques (botão direito com dois dedos, botão do meio com 3 dedos, etc). 

Espero ter ajudado em algo.

**Comment by felipefialho** [comment]
Linkando: #237

**Comment by didiraja** [comment]
Foque em Processador(compilação e localhost) e SSD, esse faz uma diferença danada tanto nos builds quando na busca de arquivos, principalmente se tratando de projetos grandes.

**Comment by thiagok** [comment]
Antes de mais nada, considere comprar com SSD. É mais caro, mas vale muito a pena. Tem dado conta em Macbooks de 2011 sem nenhum problema.

**Comment by alexsandro-xpt** [comment]
@thiagok meu Macbook Pro de 2011 pifou ontem... problema na GPU, ele liga mas trava no login sem falar que a tela fica com linhas horizontais.... Era e é um i7 com 16 de RAM. Ainda não sei o que faço.

**Comment by willycamargo** [comment]
**Notebook Dell XPS 15 2012 - 15" Full HD**

Settings: i7 2.2ghz, 4gb RAM, 500gb HD

Comprei o notebook novo e paguei cerca de 2.500 na época (com upgrade de bateria)
O notebook é enorme e difícil de portar para qualquer lugar (2,78 kg). Mas atendia minhas necessidades e com certeza foi um ótimo _Custo x Beneficio_. Consegui estudar e trabalhar muito nele.

Usei ubuntu e usava uma maquina virtual com windows para softwares Adobe
Trabalhar com 4gb era impossível. A maquina literalmente travava e desligava com tarefas simples. Upgrade para 8gb resolveu meu problema.

Foram 3 anos com ele sem problemas graves além de alguns contratempos com setup no ubuntu. vendi ele por 1.500 em ótimo estado e até hoje a pessoa que comprou elogia a máquina.

----------------------------

**Macbook Air 2015 - 13"**

Settings: i7 2,2 GHz, 8gb RAM, 512gb SSD

Comprei com 2 anos de uso e paguei 3.200.
O notebook é super leve e portátil. Ao colocar na mochila nem se percebe que tem um computador ali. O carregador também é fácil de levar para um lado ou outro.

A maior vantagem é conseguir fazer tudo com um único sistema e sem gambiarra.
O setup de software é simples e direto, com muito conteúdo online para quando tiver qualquer dúvida. Em menos de 2h eu consigo deixar pronto com todos os softwares e configurações para desenvolver.

Os 8gb são suficientes para mim rodando duas aplicações front + 1 API back. Só sinto o computador gargalar quando abro muita coisa nele, mas dai é apenas questão de saber se organizar. Confesso que seria luxo dizer que ele não da conta. Ele é suficiente para tudo que preciso fazer.

Estou há 1 ano com esse macbook e já tenho comprador pra ele pelo exato mesmo valor que paguei (3.200)

----------------------------

**Macbook Pro 2015 - 15" Retina**

Settings: i7 2,2 GHz, 16gb RAM, 256gb SSD

Comprei com 1 ano de uso e paguei 6.000.
O computador é menor e mais leve que o dell que tive, porém ainda assim é muito pesado e grande. Não gosto de carregar ele, então só faço quando realmente preciso.

Tirando a portabilidade as vantagens são exatamente as mesmas do macbook air acima.

As configurações do computador são excedentes para o que faço no dia a dia. Tudo responde muito rápido e não sinto necessidade de nada. As vezes me perco com um photoshop ou illustrator que deixei aberto há dias ali e a maquina não trava.

Estou há 2 anos com esse macbook e tive problemas com a tela dele (staingate), apesar de fora da garantia a Apple tinha um recall exclusivo para esse problema e o macbook ficou 1 mês na assistência até eu pegar novamente (foi quando comprei o macbook air 13).

Já cheguei a negociar ele e teria vendido por 6.500 se não tivesse desistido da venda. O fato é que sou apaixonado pela tela de retina e não vejo tanta necessidade de portabilidade, portanto prefiro ele quanto ao macbook air.

----------------------------

Acho que a resposta de "custo x beneficio" não é tão exata quanto podemos imaginar. Isso depende muito do que você busca, quais suas condições financeiras e o quão cuidadoso e paciente é.

Se você tem um pouco mais de condição e for uma pessoa extremamente cuidadosa eu recomendo ir atras de um macbook seminovo porque mesmo após alguns anos ainda conseguirá vende-lo por um bom preço. Temos um mercado bom para produtos apple usados no Brasil.

Se tem menos condições, vá atras de um dell ou lenovo com configurações medianas. Tenha paciência quando a máquina não te entregar todo desempenho que precisa. Isso não fará de você pior.

Pra ser sincero eu acho que uma das maiores vantagens de um macbook é conseguir revende-lo sem desvalorizar praticamente nada. O que é mais difícil com outras marcas.

**Comment by matheus-rosin** [comment]
@willycamargo então, é bem isso que eu tava pensando, em pegar um macbook usado... Só que ficam duas dúvidas principais:

Eu não manjo nada do sistema que vem nele (até hoje, só usei windows -- acho que mal vi um mac na frente haha), então minha questão é: tem que levar em consideração a versão do OS que vem nele? Tem perigo de o laptop ficar "obsoleto" e não atualizar mais?

E outra dúvida mais prática: onde comprar usado? Teria que ser de conhecido, ou existe algum meio mais confiável em que posso achar? Mercado Livre? ...

**Comment by willycamargo** [comment]
Eu considero o sistema simples e fácil de se adaptar. O sistema da praticamente tudo na sua mão. Considero o sistema com melhor usabilidade do mercado.

Não sei como as maquinas mais antigas estão se comportando com as versões mais atuais dos Mac OS, mas meus dois notebooks estão rodando a versão mais atual (Mojave) sem problemas nenhum. Acredito que nenhum macbook com SSD sofra com as atualizações atuais. Sempre espero receber uma notificação de atualização e faço a atualização da forma mais simples possível (apertando o botão de continuar e esperando a barrinha carregar hahaha).

Mantenho o computador sempre limpo e com o mínimo de software instalados, então nunca senti a necessidade de formatar. Mas ja formatei um em uma empresa que trabalhava e foi muito simples. No próprio site da apple encontra um step-by-step de como fazer.

Sobre onde comprar: eu recomendo comprar de mão a mão. Melhor para ver o estado do computador, do carregador etc. Um bom lugar para pesquisar seria OLX. Bom que foge da taxa do ML e tem a certeza de estar comprando algo bom.

Recomendações:

- Verifique se a maquina possuí riscos, marcas de tombo ou marcas de liquido.
- Ainda fechado, vire a maquina de ponta cabeça e verifique os parafusos. Veja se esta faltando algum e se eles tem marcas de que foram abertos. Se possuem marcas pergunte para o dono por que a maquina foi aberta.
- Verifique as entradas dos cabos e teste se puder (energia, usb, sd card etc)
- Verifique o estado da tela, se possui riscos e, se for macbook air, se a moldura em volta do vidro esta descolando (isso acontece com quem costuma abrir e fechar a tampa de qualquer jeito)
- Após ligar, confira as configurações para ter certeza que é o que anunciaram
- Verifique a quantidade de ciclos de bateria (pede para o dono e se ele não souber, de um google sobre como fazer isso)

Um detalhe importante é que dizem que com aproximadamente 1.000 ciclos a bateria do mac já esta com o pé na cova e é normal os proprietários trocarem a bateria por uma paralela, o que é compreensível dado o valor de uma bateria original. Porém as baterias paralelas variam muito de qualidade. Algumas são tão boas quanto as originais e outras são péssimas. Eu evitaria comprar um macbook que teve bateria substituída. Eu preferiria comprar com a bateria estragada com desconto e eu mesmo levar em algum lugar confiável para substituir. Para saber se foi substituída basta ver o número de ciclos da bateria e ver se condiz com o estado da maquina. Maquina com muitas marcas de uso e teclas desgastadas com bateria com poucos ciclos (menos de 250) pode indicar duas coisas: ou a bateria foi trocada ou o computador foi usado muito direto na tomada. Para julgar isso de forma mais objetiva a dica dos parafusos podem te ajudar.

Sei que pode parecer um "puta trabalho" apenas para comprar um computador. Mas pelo menos pra mim valeu a pena. Não tinha dinheiro para comprar um novo (11mil) e tinha tempo para poder fazer as pesquisas.

**Comment by andersoonweb** [comment]
Tive problemas com o meu semana passada. É um Asus i3, 6GB de RAM e não tinha SSD, estava pensando em colocar. Eu precisava de um upgrade, porém, eu estava conseguindo sobreviver com ele. Utilizava Windows em Dual Boot com Ubuntu.
Agora, estou na busca de comprar um, cogitei em comprar um mac, mas não gosto de aparelhos usados e também vi que não vale a pena pra mim, não cabe no meu orçamento. E tudo isso pra rodar Unix + Adobe.

Eu vou de uma máquina mais completa agora e fazer alguns upgrade com o tempo. Tô acompanhando o preço do Asus VivoBook X510UR - i7 8° geração + 8GBRAM + 1TB HD e com slot que suporta M2 - Vou ter que colocar um SSD futuramente. Pretendo rodar Windows e Linux também. Esse aparelho parece dar conta do que quero fazer. Em reviews sempre falam que é o melhor custo benefício.

Fica um conselho meu: Olhe bem o que você vai comprar. Um mac pode valer muuuuuuuito a pena pra você, mas também pode ser apenas status e pagar por um aparelho razoável enquanto pode ter um bom. Esse lance de gambiarra vai de cada um, talvez pra você rodar um VM seja extremamente normal - como é pra o Fábio Akita, sugiro acompanhar alguns posts dele. Assim como, a questão de RAM, eu sei que existe diversas coisas extremamente pesada, mas se houver uma organização da sua parte dá pra reduzir e conseguir trabalhar.
